src/Type/Union/MenuItemObjectUnion.php

```diff
@@ -52,7 +52,7 @@
 				return Types::term_object( $object->taxonomyName );
 			}
 
-			return null;
+			return apply_filters( 'resolve_menu_item_type', null, $object );
```

Comment:
🤔 This seems like something we need to resolve centrally. This _should_ be a filter applied to _all_ Union Types. We should consider applying this filter centrally within WPUnionType class, instead of adding a filter for each Union. 

Can you also elaborate on what this change does and why it's necessary?

Reply:
Naw, I would agree if all unions resolved the same way but they are each very different in how they operate, however `MenuItemObjectUnion` targets post-types using `get_post_types( [ 'show_in_graphql' => true, 'show_in_nav_menus' => true ] )`, and I think that's the right way to do it. In [PostObjectUnion](https://github.com/wp-graphql/wp-graphql/blob/develop/src/Type/Union/PostObjectUnion.php#L8), I think is `get_post_types( [ 'show_in_graphql` => true ] )` should be used instead of `WPGraphQL::get_allowed_post_types()`, but that's the only change I would make.

Reply:
Well, where you placed this filter is within the resolver, which is decoupled from the use of `get_post_types()` in this file. 

I think if we add a filter to pass over all `resolveType` callbacks for `all` Unions, then any union could be hooked into, not _just_ this particular union.

Reply:
@kidunot89 ok, so I opened a PR to your fork here showing what I was thinking: https://github.com/kidunot89/wp-graphql/pull/2

This moves the filter centrally so _any_ Union in the entire Schema can benefit from this filter. 

Then, in your codebase, all you would need to do is change the name of the filter from `resolve_menu_item_type` to `graphql_union_resolve_type` and it should work the same, but with the added benefit that all Unions throughout the Schema are also filterable.

Reply:
@kidunot89 let me know what you think here.

Reply:
Merged